lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1421,14 +1421,24 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     return mapValue(V, BI);
   }
 
-  if (SelectInst *Sel = dyn_cast<SelectInst>(V))
-    return mapValue(
-        V,
-        BM->addSelectInst(
-            transValue(Sel->getCondition(), BB),
-            transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer),
-            transValue(Sel->getFalseValue(), BB, true, FuncTransMode::Pointer),
-            BB));
+  if (SelectInst *Sel = dyn_cast<SelectInst>(V)) {
+    auto CV = transValue(Sel->getCondition(), BB);
+    auto TV = transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer);
+    auto FV = transValue(Sel->getFalseValue(), BB, true, FuncTransMode::Pointer);
+
+    // LLVM can generate vector selects with a bool selector operand,
+    // This isn't supported until SPIRV 1.4. For now translate them
+    // into vector selects and hope the backend can undo this.
+    if (!CV->getType()->isTypeVector() && TV->getType()->isTypeVector()) {
```

Comment:
One more thing. Do I get it right that without this PR it works correctly if we want to generate SPIR-V 1.4 or above. If so, I was going to suggest:
```suggestion
    if (!CV->getType()->isTypeVector() && TV->getType()->isTypeVector() &&
        !BM->isAllowedToUseVersion(VersionNumber::SPIRV_1_4)) {
```
But after https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/505/commits/1eb284b1e0b2c18ee44be6717cb11fd1a00d9531 from @bashbaug, the maximum supported version is 1.3
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/9326deff2916f9f566aadf5cedeb58f8d65e945e/include/LLVMSPIRVOpts.h#L49-L60

Does anybody knows if we can populate the enum with `SPIRV_1_4` at least?

Reply:
@airlied Could you please add `SPIRV_1_4 = 0x00010400,` to this enum, update `MaximumVersion` and apply my suggestion above.
Other than that, LGTM. Thanks!